src/simplewallet/simplewallet.cpp

```diff
@@ -1136,6 +1151,17 @@ bool simple_wallet::init(const boost::program_options::variables_map& vm)
         return false;
       }
     }
+    if (!m_restore_height)
+    {
+      std::string heightstr = command_line::input_line("Restore from specific blockchain height (optional, default 0): ");
+      try {
+        if (heightstr.size())
+          m_restore_height = boost::lexical_cast<uint64_t>(heightstr);
+      }
+      catch (boost::bad_lexical_cast &) {
+        fail_msg_writer() << tr("bad m_restore_height parameter:") << " " << heightstr;
```

Comment:
Do you want to return false here ? Else, print a "defaulting to scanning from start" to avoid it looking like an actual unhandled error.